suite-native/module-send/src/components/SendFeesForm.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useWatch } from 'react-hook-form';
+
+import { isRejected } from '@reduxjs/toolkit';
+import { useNavigation } from '@react-navigation/native';
+
+import { VStack, Text, Box } from '@suite-native/atoms';
+import { Form, useForm } from '@suite-native/forms';
+import {
+    AccountKey,
+    GeneralPrecomposedLevels,
+    PrecomposedTransactionFinal,
+} from '@suite-common/wallet-types';
+import { AccountsRootState, selectAccountByKey } from '@suite-common/wallet-core';
+import {
+    SendStackParamList,
+    SendStackRoutes,
+    StackNavigationProps,
+} from '@suite-native/navigation';
+import { useToast } from '@suite-native/toasts';
+import { Translation } from '@suite-native/intl';
+
+import { signTransactionFeeLevelSigningThunk } from '../sendFormThunks';
+import { SendFeesFormValues, sendFeesFormValidationSchema } from '../sendFeesFormSchema';
+import { FeesFooter } from './FeesFooter';
+import { FeeOptionsList } from './FeeOptionsList';
+
+type SendFormProps = {
+    accountKey: AccountKey;
+    feeLevels: GeneralPrecomposedLevels;
+};
+
+type SendFeesNavigationProps = StackNavigationProps<SendStackParamList, SendStackRoutes.SendReview>;
+
+const DEFAULT_FEE = 'normal';
+
+export const SendFeesForm = ({ accountKey, feeLevels }: SendFormProps) => {
+    const dispatch = useDispatch();
+    const { showToast } = useToast();
+    const navigation = useNavigation<SendFeesNavigationProps>();
+
+    const account = useSelector((state: AccountsRootState) =>
+        selectAccountByKey(state, accountKey),
+    );
+
+    const form = useForm<SendFeesFormValues>({
+        validation: sendFeesFormValidationSchema,
+        defaultValues: {
+            feeLevel: DEFAULT_FEE,
+        },
+    });
+    const { handleSubmit, control } = form;
+
+    const selectedFeeLevel = useWatch({ control, name: 'feeLevel' });
+    const feeLevelTransaction = feeLevels[selectedFeeLevel] as PrecomposedTransactionFinal;
+
+    if (!account) return;
+
+    const handleNavigateToReviewScreen = handleSubmit(async () => {
+        navigation.navigate(SendStackRoutes.SendReview, { accountKey });
```

Comment:
move this navigation after `isRejected` check of `signTransactionFeeLevelSigningThunk` or it app will navigate and crash in case `signTransactionFeeLevelSigningThunk` rejects.

Reply:
Make sense on the first look, but unfortunately we can not do that. While is the `signTransactionFeeLevelSigningThunk` executing, the user needs to be already on the `SendReviewScreen` where he can see the individual steps of the review and their status. 

If the navigation would happen after the dispatch of `signTransactionFeeLevelSigningThunk` the Trezor device would start to display inputs for the review, but the app would not correspond with that.

In case it fails, it correctly returns to the beginning of the flow (account selection screen). I tested that. Or did it not work for you? If no, please let me know how to replicate that.

Reply:
As we discussed the issue here was a validation that shouldnt let it pass and it fill be done in following PRs. Resolving